### ISSUE TITLE
CIELUV does not guarantee constant hue when changing saturation

### ISSUE THREAD

**Issue by snipstruis** [hsluv/hsluv#52]
Hi! First I'd like to say I really like HSLuv! it's way better than HSL and more practical than CIELUV.

There is only one nearly-unfixable problem: CIELUV does not guarantee constant hue when changing saturation.

This has two effects:
1) (I don't see this myself) different colors with the same saturation may appear to have slightly different saturation.
2) (the real problem) moving towards or away from the center (i.e. changing the saturation) may also change the percepted hue.

Problem number 2 is most noticable with blues.
If you look at #0000FF blue and slide it to about 40% saturation you get #454587, which appears too purple (subjectively, I think it should be more like #34477f)

Unfortunately, when sliding with the lightness-slider, HSLuv does a lot of sliding towards or away from the center in order to keep the color in gammut. This means that for HSLuv chages in lightness can ALSO change the hue, so problem number 2 is an even bigger problem for HSLuv.

I still think HSLuv is the most practical system as of yet, but it has this almost-unsolvable problem.

I hope I wasn't too much of a downer,
Snipstruis

(ps this probbably also explains issue #41)

**Comment by 370417** [comment]
Yes, CIELAB and CIELUV perform badly with blues. People aren't too sure why: [A hypothesis regarding the poor blue constancy of CIELAB](https://www.researchgate.net/publication/229745529_A_hypothesis_regarding_the_poor_blue_constancy_of_CIELAB).

That said, there are other perceptual color spaces do a better job of preserving hue across saturation. Recently, Björn Ottosson released [Oklab](https://bottosson.github.io/posts/oklab/), which does just that. Also of note is [IPT](https://scholarworks.rit.edu/theses/2858/), which apparently does great with hue but suffers in lightness and chroma.

Using a more recent color space might help alleviate this "almost-unsolvable" problem. Oklab has several implementations. After a quick search, IPT seems much harder to find.
